javascript/ql/src/semmle/javascript/dataflow/Configuration.qll

```diff
@@ -1260,9 +1259,11 @@ private predicate storeToLoad(
   DataFlow::Node pred, DataFlow::Node succ, DataFlow::Configuration cfg, PathSummary oldSummary,
   PathSummary newSummary
 ) {
-  exists(string prop, DataFlow::Node base |
-    reachableFromStoreBase(prop, pred, base, cfg, oldSummary) and
-    loadStep(base, succ, prop, cfg, newSummary)
+  exists(string storeProp, string loadProp, DataFlow::Node storeBase, DataFlow::Node loadBase, PathSummary s1, PathSummary s2 |
+    storeStep(pred, storeBase, storeProp, cfg, s1) and
+    reachableFromStoreBase(storeProp, loadProp, storeBase, loadBase, cfg, s2) and
+    oldSummary = s1.appendValuePreserving(s2) and
+    loadStep(loadBase, succ, loadProp, cfg, newSummary)
```

Comment:
It appears to me that we can track more flow with this change.   
A test demonstrating this new flow would be nice.

Reply:
It was meant to be a semantics-preserving change.

Previously `rhs` took on the role of `startProp` since it was specific to a store. Now that we track the base node instead, we need to remember what `startProp` was. It still seems to be a win in cases where there are a large number of stores of the same property name.

Reply:
I think I get it now :+1: